Content.Server/SimpleStation14/Jukebox/JukeboxSystem.cs

```diff
@@ -0,0 +1,308 @@
+
+            Play(jukeBox, jukeboxComp, toPlay);
+
+            return;
+        }
+
+        UpdateState(jukeBox, jukeboxComp);
+    }
+
+    /// <summary>
+    ///     Pauses the currently playing song and sets the Jukebox to a paused state.
+    /// </summary>
+    /// <remarks>
+    ///     See <see cref="Stop"/> to stop the Jukebox instead.
+    ///     Pausing a Jukebox will allow it to remember its paused state, even if it gets stopped later.
+    /// </remarks>
+    public void Pause(EntityUid jukeBox, JukeboxComponent jukeboxComp)
+    {
+        if (jukeboxComp.Paused || !jukeboxComp.Playing)
+            return;
+
+        jukeboxComp.Paused = true;
+
+        Stop(jukeBox, jukeboxComp);
+    }
+
+    /// <summary>
+    ///     Unpauses the Jukebox and resumes playing the current song from where it was paused.
+    /// </summary>
+    public void UnPause(EntityUid jukeBox, JukeboxComponent jukeboxComp)
+    {
+        if (!jukeboxComp.CanPlay)
+            return;
+
+        if (!jukeboxComp.Paused)
+            return;
+
+        jukeboxComp.Paused = false;
+
+        TryRestart(jukeBox, jukeboxComp);
+    }
+
+    /// <summary>
+    ///     Plays a song in the Jukebox using the provided <see cref="JukeboxTrackPrototype"/> and optional offset.
+    ///     If the Jukebox is already playing a song, it will be stopped and replaced with the new song.
+    ///     To queue a song instead, see <see cref="TryQueueSong"/>.
+    /// </summary>
+    /// <param name="song">The JukeboxTrackPrototype representing the song to be played.</param>
+    /// <param name="offset">The optional offset from the start of the song to begin playing from.</param>
+    public void Play(EntityUid jukeBox, JukeboxComponent jukeboxComp, string song, TimeSpan offset = new TimeSpan())
+    {
+        if (!_prototype.TryIndex<JukeboxTrackPrototype>(song, out var songPrototype))
+        {
+            Logger.Error($"Jukebox track prototype {song} not found!");
+
+            Skip(jukeBox, jukeboxComp);
+            return;
+        }
+
+        Play(jukeBox, jukeboxComp, songPrototype, offset);
+    }
+
+    /// <inheritdoc cref="Play(EntityUid, JukeboxComponent, string, TimeSpan)"/>
+    /// <remarks>
+    ///     Directly takes a <see cref="JukeboxTrackPrototype"/> instead.
+    /// </remarks>
+    public void Play(EntityUid jukeBox, JukeboxComponent jukeboxComp, JukeboxTrackPrototype song, TimeSpan offset = new TimeSpan())
+    {
+        if (!jukeboxComp.CanPlay)
+            return;
+
+        Clean(jukeBox, jukeboxComp);
+
+        jukeboxComp.Paused = false;
+
+        jukeboxComp.Playing = true;
+
+        jukeboxComp.CurrentlyPlayingTrack = song;
+
+        jukeboxComp.FinishPlayingTime = _timing.CurTime + song.Duration - offset;
+
+        jukeboxComp.CurrentlyPlayingStream = _audio.Play(song.Path, Filter.Broadcast(), jukeBox, true, AudioParams.Default.WithPlayOffset((float)offset.TotalSeconds));
```

Comment:
This variable should be managed on the client, and the Jukebox should simply sync the information that it *is* playing. The Client can then handle playing the song, calculating the offset by time remaining, time started, etc.. This way, new players will have their Jukeboxes synced with current players.
I also don't know how the current system will play with PVS.